backend/routers/transcribe_v2.py

```diff
@@ -0,0 +1,368 @@
+        # stt_service: STTService = STTService.deepgram,
+):
+    print('websocket_endpoint', uid, language, sample_rate, codec, channels, include_speech_profile, new_memory_watch)
+
+    # Not when comes from the phone, and only Friend's with 1.0.4
+    if language == 'en' and sample_rate == 16000 and codec == 'opus':
+        stt_service = STTService.soniox
+    else:
+        stt_service = STTService.deepgram
+
+    try:
+        await websocket.accept()
+    except RuntimeError as e:
+        print(e)
+        return
+
+    # Initiate a separate vad for each websocket
+    w_vad = webrtcvad.Vad()
+    w_vad.set_mode(1)
+
+    # Stream transcript
+    loop = asyncio.get_event_loop()
+
+    # Soft timeout, should < MODAL_TIME_OUT - 3m
+    timeout_seconds = 420  # 7m
+    started_at = time.time()
+
+    def _get_in_progress_memory(segments: List[dict]):
+
+        memory_id = redis_db.get_in_progress_memory_id(uid)
+        existing = None
+
+        if memory_id:
+            existing = memories_db.get_memory(uid, memory_id)
+            if existing and existing['status'] != 'in_progress':
+                existing = None
+
+        if not existing:
+            existing = memories_db.get_in_progress_memory(uid)
+
+        if existing:
+            print('_get_in_progress_memory existing', existing['id'])
+            memory = Memory(**existing)
+            memory.transcript_segments = TranscriptSegment.combine_segments(
+                memory.transcript_segments, [TranscriptSegment(**segment) for segment in segments]
+            )
+            redis_db.set_in_progress_memory_id(uid, memory.id)
+            return memory
+        started_at = datetime.now(timezone.utc) - timedelta(seconds=segments[0]['end'] - segments[0]['start'])
+        memory = Memory(
+            id=str(uuid.uuid4()),
+            uid=uid,
+            structured=Structured(),
+            language=language,
+            created_at=started_at,
+            started_at=started_at,
+            finished_at=datetime.now(timezone.utc),
+            transcript_segments=[TranscriptSegment(**segment) for segment in segments],
+            status=MemoryStatus.in_progress,
+        )
+        print('_get_in_progress_memory new', memory)
+        memories_db.upsert_memory(uid, memory_data=memory.dict())
+        redis_db.set_in_progress_memory_id(uid, memory.id)
+        return memory
+
+    async def memory_creation_timer():
+        try:
+            await asyncio.sleep(120)
+            await _create_memory()
+        except asyncio.CancelledError:
+            pass
+
+    memory_creation_task = None
+    segment_start = None
+
+    def stream_transcript(segments, _):
+        nonlocal websocket
+        nonlocal segment_start
+        nonlocal memory_creation_task
+
+        if not segments or len(segments) == 0:
+            return
+
+        # Align the start, end segment
+        if not segment_start:
+            segment_start = segments[0]["start"]
```

Comment:
if we want to keep the socket after memory created (without forcing it close from the app), the segment_start should be reset(to None) to adapt with the new segments so that we don't need to reset the STT sockets.